pkgs/build-support/php/build-php.nix

```diff
@@ -0,0 +1,106 @@
+{ stdenvNoCC, lib, writeTextDir, composer, php }@inputs:
+
+{ pname
+, version
+, src
+, composerLock ? "__unset"
+, vendorDir ? "vendor"
+, vendorHash
+, php ? inputs.php
+, composer ? inputs.composer
+, ...
+}@args:
+
+let
+  vendor = stdenvNoCC.mkDerivation (finalAttrs: {
+    inherit src version;
+
+    pname = "${pname}-vendor";
+
+    nativeBuildInputs = [
+      composer
+    ];
+
+    strictDeps = true;
+
+    configurePhase = ''
+      runHook preConfigure
+
+      if [ -f ${src}/composer.lock ]
+      then
+        cp -f ${src}/composer.lock composer.lock
+      fi
```

Comment:
This should be already done by the `unpackPhase`. And would not work for tarball sources.

Reply:
I did it in a `postUnpack` phase.

Reply:
Couldn't get it working properly. In the end, I moved everything in `buildPhase`.